### ISSUE TITLE
Allow resources without a listView in the Menu

### ISSUE THREAD

**Issue by DjebbZ** [marmelab/react-admin#6562]
**Is your feature request related to a problem? Please describe.**
First of all, not sure if it's a feature request, a bug report or a support question. Sorry, I'm a bit confused.

The problem I have is that some resources we have don't have a list view, the resource itself is just an editable object (think a config object for example). I haven't find a way simple way to add such resources in the Menu without doing hacks and what I think is useless and cumbersome plumbing.

What we do is the following, and the problems we face (the example below consider a resource with Show and Edit views)  :
- Create a `<Resource>` with a dummy list component so that it appears in the menu:
  ``` jsx
  export const DummyListPage = (props) => {
    return (
        <List {...props}>
            <Datagrid />
        </List>
    );
  }; 
  ``` 
- Add `customRoutes` in the `<Admin>` component that overrides the list URL that's supposed to be used for the List view
- In this component, fetch the data with a manual `useQuery` and instantiate 2 `<Route>` wrapped with a `<Switch>` for the  Show and Edit views, which get passed the fetched record:
  ``` jsx
  <Switch>
    <Route path="/my-resource/my-resource-id/show"><MyShowComponent record={data} /></Route>
    <Route path="/my-resource/my-resource-id"><MyEditComponent record={data} /></Route>
  </Swtich>
  ```
- The custom Show and Edit components use react-admin's `<Show>` and `<Edit>` component, but we manually pass the `basePath`, `resource`, `record` and the `id`.

**Describe the solution you'd like**
A new option in the `<Resource>` component called i.e. `defaultView` with values `list | show | create | edit`. If I use `show` for example,  the Show component I write is called normally with the intended props. Same for `edit` etc. We wouldn't need custom routes, we wouldn't need to manually call `useQuery`, we wouldn't need a DummyListPage, we would declare the Show and Edit normally in the `<Resource>` and everything would work as advertised in the documentation. I feel like all the plumbing/customization I'm doing doesn't bring value because it's not related to a custom feature I'm implementing but a limitation of React-admin.

**Describe alternatives you've considered**
I've also tried to add the Show and Edit normally in `<Resource>` and to replace the `<Route>` in `customRoutes` with a `<Redirect from="/path/to/list" to="path/to/show" />`, but it seems to create an infinite loop of `setState` and displays an error.

I've also tried to add the Show and Edit normally in `<Resource>`, keep the `<Switch />` but I get an empty Show page without an error.

**Additional context**
Thanks in advance for taking the time to read this issue!

**Comment by djhi** [comment]
Wouldn't a [custom menu](https://marmelab.com/react-admin/Theming.html#using-a-custom-menu) that would target custom routes be enough?

**Comment by fzaninotto** [comment]
Yes, Gildas is right. If you want to customize the menu, don't mess with the resources. Use a custom menu, as explained in the Theming documentation. 

https://marmelab.com/react-admin/Theming.html#using-a-custom-menu

**Comment by DjebbZ** [comment]
Indeed, I manage to get something almost perfect and boilerplate-free without changing our already existing custom menu. Just need to find where the `list` requirement live in RA's code, can point it to me ?

**Comment by fzaninotto** [comment]
What do you mean by `list` requirements? Whay ar eyou trying to achieve that isn't explained in the "Using a Custom Menu" documentation?

**Comment by DjebbZ** [comment]
Ok, I sorted it out 100%. We started without a custom menu, and got used to the `DummyListPage` trick to make entries in the left menu for resources without `list` page. We had to use `customRoutes`, `<Switch>` and all I described above. We didn't realize that simply declaring the `show`/`edit`/`create` pages and adding a custom menu entry is enough to be able to use normally the `Show`, `Create`, `Edit` components (that call the controller which interacts with the dataProvider etc.). We switched to a custom menu some time ago but didn't take the time to investigate all of this.

I just declared the pages, added a custom menu entry , and now it's really just "normal" React-admin, as in the documentation. RA is even more awesome to my eyes now. Thanks for the guidance!
